src/clr-angular/forms/combobox/providers/option-selection.service.ts

```diff
@@ -1,53 +1,75 @@
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
 import { Injectable } from '@angular/core';
-import { Observable, Subject } from 'rxjs';
+import { Observable, Subject, BehaviorSubject } from 'rxjs';
 
 import { ClrOption } from '../option';
+import { MultiSelectComboboxModel } from '../model/multi-select-combobox.model';
+import { SingleSelectComboboxModel } from '../model/single-select-combobox.model';
+
+export type SelectionModel<T> = SingleSelectComboboxModel<T> | MultiSelectComboboxModel<T>;
 
 @Injectable()
 export class OptionSelectionService<T> {
-  private currentValue: T;
-  private currentOption: ClrOption<T>;
+  public selectionModel: SelectionModel<T>;
+  public currentOption: ClrOption<T>;
+  public loading: boolean = false;
+  public displayField: string;
+  private _currentInput: string = '';
+  get currentInput(): string {
+    return this._currentInput;
+  }
+  set currentInput(input) {
+    this._currentInput = input;
+    this._inputChanged.next(input);
+  }
+  private _inputChanged: BehaviorSubject<string> = new BehaviorSubject('');
+  inputChanged = this._inputChanged.asObservable();
 
-  private _valueChanged: Subject<T> = new Subject<T>();
+  private _selectionChanged: Subject<SelectionModel<T>> = new Subject<SelectionModel<T>>();
 
   // This observable is for notifying the ClrOption to update its
   // selection by comparing the value
-  get valueChanged(): Observable<T> {
-    return this._valueChanged.asObservable();
+  get selectionChanged(): Observable<SelectionModel<T>> {
+    return this._selectionChanged.asObservable();
   }
 
-  // NOTE: Currently only handles single selection
-  // TODO: Add suport for trackBy and compareFn
-  updateSelection(value: T): void {
-    // NOTE: Currently we assume that no 2 options will have the same value
-    // but Eudes and I discussed that this is a possibility but we will handle
-    // this later
-    if (this.currentValue && this.currentValue === value) {
+  setSelected(item: T, selected: boolean) {
+    // if new item is null (or undefined) return
+    if (!item) {
       return;
     }
-    this.currentValue = value;
-    this._valueChanged.next(value);
-  }
 
-  private _renderSelectionChanged: Subject<ClrOption<T>> = new Subject<ClrOption<T>>();
+    const oldValue = this.selectionModel.model;
+
+    this.selectionModel.setSelected(item, selected);
 
-  // This observable is to notify the ClrCombobox component to render
-  // a new Option on the Input
-  get renderSelectionChanged(): Observable<ClrOption<T>> {
-    return this._renderSelectionChanged.asObservable();
+    // only emit changes if the value actually changed
+    if (this.selectionModel.model !== oldValue) {
+      this._selectionChanged.next(this.selectionModel);
+    }
+  }
+
+  get multi(): boolean {
```

Comment:
nit: is there any way this can be made the same as the attribute its used on? e.g `multiselectable`

Reply:
`multiselectable` it will be then